lib/Analysis/Membar.cpp

```diff
@@ -138,14 +138,19 @@ void MembarAnalysis::update(Operation *op, BlockInfo *blockInfo,
     for (Value value : op->getOperands()) {
       for (auto bufferId : allocation->getBufferIds(value)) {
         if (bufferId != Allocation::InvalidBufferId) {
-          if (isa<triton::gpu::AsyncCopyGlobalToLocalOp>(op)) {
-            // Global -> shared memory
-            curBlockInfo.syncWriteIntervals.insert(
-                allocation->getAllocatedInterval(bufferId));
-          } else {
-            // ConvertLayoutOp: shared memory -> registers
-            curBlockInfo.syncReadIntervals.insert(
-                allocation->getAllocatedInterval(bufferId));
+          MemoryEffectOpInterface memIF(op);
+          SmallVector<MemoryEffects::EffectInstance> effects;
+          memIF.getEffects(effects);
+          for (const auto &effect : effects) {
+            if (isa<MemoryEffects::Write>(effect.getEffect())) {
+              // Global -> shared memory
+              curBlockInfo.syncWriteIntervals.insert(
+                  allocation->getAllocatedInterval(bufferId));
+            } else if (isa<MemoryEffects::Read>(effect.getEffect())) {
+              // ConvertLayoutOp: shared memory -> registers
+              curBlockInfo.syncReadIntervals.insert(
+                  allocation->getAllocatedInterval(bufferId));
+            }
```

Comment:
how do we distinguish shared memory to global memory accesses?

Reply:
In the current implementation, it must have a `bufferId`, which are only managed for shared memory buffers.

Reply:
Hi @sjw36 , I think you can just add a special condition for `local_store` for now to unblock you.

``` 
if (isa<triton::gpu::LocalStoreOp>(op)) {
...
}
```

I plan to get rid of all op-specific check in membar in another PR, may take a few days.

Reply:
Will do, thanks.